src/gui/importCIFDialogFuncs.cpp

```diff
@@ -509,6 +569,48 @@ void ImportCIFDialog::on_MoietyNETARemoveFragmentsCheck_clicked(bool checked)
         createCleanedSpecies();
 }
 
+// Detect unique species in the structural species
+bool ImportCIFDialog::detectUniqueSpecies()
+{
+    // Clear any existing species
+    cifSpecies_.clear();
+
+    std::vector<int> indices(cleanedSpecies_->nAtoms());
+    std::iota(indices.begin(), indices.end(), 0);
+
+    auto idx = 0;
+
+    // Find all CIFSpecies, and their instances
+    while (!indices.empty())
```

Comment:
Let's move this declaration tighter to the `while` loop so its intended place of use is a little clearer.
```suggestion
    // Find all CIFSpecies, and their instances
    auto idx = 0;
    while (!indices.empty())
```